ios/RNVisitableView.swift

```diff
@@ -14,17 +14,34 @@ class RNVisitableView: UIView {
   @objc var onVisitProposal: RCTDirectEventBlock?
   @objc var onLoad: RCTDirectEventBlock?
   @objc var onVisitError: RCTDirectEventBlock?
+  @objc var sessionHandle: NSNumber?
+  var bridge: RCTBridge?
   
   private var controller: RNVisitableViewController?
   
+  init(bridge: RCTBridge) {
+    self.bridge = bridge
+    super.init(frame: CGRect.zero)
+  }
+
+  required init?(coder aDecoder: NSCoder) {
+    super.init(coder: aDecoder)
+  }
+  
+  public func getSession() -> Session? {
+    guard let view = self.bridge?.uiManager?.view(forReactTag: sessionHandle!) as? RNSession else {
+      print("Couldn't find session for sessionHandle:", sessionHandle!)
+      return nil
+    }
+    return view.session
```

Comment:
```suggestion
    guard let sessionHandle = sessionHandle else {
      print("Couldn't find session for nil")
      return nil
    }
    
    guard let view = self.bridge?.uiManager?.view(forReactTag: sessionHandle) as? RNSession else {
      print("Couldn't find session for sessionHandle:", sessionHandle)
      return nil
    }
    return view.session
```

Reply:
Also, I would add a check if that function was called on the main thread - you can use `Thread.isMainThread` to do it.

Reply:
Hello @lukmccall thanks for review. I was thinking about `Thread.isMainThread` and `caching` issue 🤔 but still don't really understand the issue. Can you add some more context?

Reply:
Resolved as discussed.